### PR TITLE
easyblock for VMD

### PR DESCRIPTION
**Pull request by boegel**
This is a cleaned up/enhanced version of the VMD easyblock contributed by @thiell in #879, works with VMD 1.9.3.

The CUDA/OptiX support added by @thiell should still work, but this has not been tested.

### CODE REVIEW

File: easybuild/easyblocks/v/vmd.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        # LP64: build VMD as 64-bit binary
+        # IMD: enable support for Interactive Molecular Dynamics (e.g. to connect to NAMD for remote simulations)
+        # PTHREADS: enable support for POSIX threads
+        # COLVARS: enable support for collective variables (related to NAMD/LAMMPS)
+        # NOSILENT: verbose build command
+        self.cfg.update('configopts', "LINUXAMD64 LP64 IMD PTHREADS COLVARS NOSILENT")
+
+        # add additional configopts based on available dependencies
+        for key in deps:
+            if deps[key]:
```
Comment (wpoely86) [inline easybuild/easyblocks/v/vmd.py]: why this `if`? Can it not be avoided?

File: easybuild/easyblocks/v/vmd.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            env.setvar('NUMPY_INCLUDE_DIR', out.strip())
+
+        # compiler commands
+        self.cfg.update('buildopts', 'CC="%s"' % os.getenv('CC'))
+        self.cfg.update('buildopts', 'CCPP="%s"' % os.getenv('CXX'))
+
+        # source tarballs contains a 'plugins' and 'vmd-<version>' directory
+        vmddir = os.path.join(self.cfg['start_dir'], '%s-%s' % (self.name.lower(), self.version))
+
+        # first, build plugins
```
Comment (wpoely86) [inline easybuild/easyblocks/v/vmd.py]: you have to explain this one

File: easybuild/easyblocks/v/vmd.py (hunk 1 above)
Comment (boegel) [inline easybuild/easyblocks/v/vmd.py]: some dependencies are optional, so some of these may be `None`

File: easybuild/easyblocks/v/vmd.py (hunk 2 above)
Comment (boegel) [inline easybuild/easyblocks/v/vmd.py]: well, I can provide a reference to the VMD docs, but that's how it should be done, see http://www.ks.uiuc.edu/Research/vmd/doxygen/compiling.html or http://robinbetz.com/blog/2015/01/08/compiling-vmd-with-python-support/

**Comment by boegel** [comment]
Tested with https://github.com/hpcugent/easybuild-easyconfigs/pull/4391, good to go in as revised version of #879. Thanks for the review @wpoely86!

**Pull request merged**